src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.OSX/X509Pal.cs

```diff
@@ -31,15 +33,27 @@ private partial class AppleX509Pal : ManagedX509ExtensionProcessor, IX509Pal
                     switch (oid.Value)
                     {
                         case Oids.RsaRsa:
+                            if (key.IsInvalid)
+                            {
+                                throw Interop.AppleCrypto.CreateExceptionForOSStatus(errSecUnsupportedKeyFormat);
+                            }
                             return new RSAImplementation.RSASecurityTransforms(key);
                         case Oids.DsaDsa:
-                            if (key.IsInvalid) 
+                            if (key.IsInvalid)
                             {
                                 // SecCertificateCopyKey returns null for DSA, so fall back to manually building it.
                                 return DecodeDsaPublicKey(encodedKeyValue, encodedParameters);
                             } 
                             return new DSAImplementation.DSASecurityTransforms(key);
                         case Oids.Ecc:
+                            if (key.IsInvalid)
+                            {
+                                throw Interop.AppleCrypto.CreateExceptionForOSStatus(errSecUnsupportedKeyFormat);
+                            }
+                            else if (Interop.AppleCrypto.EccGetKeySizeInBits(key) == 0)
+                            {
+                                throw Interop.AppleCrypto.CreateExceptionForOSStatus(errSecUnsupportedKeySize);
```

Comment:
I believe that the zero now is due to our code being the limiting factor, a comment for that would be good.

Reply:
It's not only the CoreFX code, but also the fact that for 224-bit keys the native Apple crypto code seems to fail in unexpected ways:

```
Interop+AppleCrypto+AppleCFErrorCryptographicException : The operation couldn’t be completed. (com.apple.security.transforms.error error 20 - ABORT has been sent to the transform (Error Domain=Internal CSSM error Code=-25304 "Internal error #ffff9d28 at VerifyTransform_block_invoke /BuildRoot/Library/Caches/com.apple.xbs/Sources/Security/Security-58286.200.178/OSX/libsecurity_transform/lib/SecSignVerifyTransform.c:540" UserInfo={NSDescription=Internal error #ffff9d28 at VerifyTransform_block_invoke /BuildRoot/Library/Caches/com.apple.xbs/Sources/Security/Security-58286.200.178/OSX/libsecurity_transform/lib/SecSignVerifyTransform.c:540, Originating Transform=CoreFoundationObject}))
```

I agree a comment is warranted.